crates/ast/src/source_atom_set.rs

```diff
@@ -197,29 +190,20 @@ impl<'alloc> SourceAtomSet<'alloc> {
     // it with the result of this method.
     pub fn new_uninitialized() -> Self {
         Self {
-            atoms: Vec::new(),
-            atom_indices: HashMap::new(),
+            atoms: IndexSet::new(),
         }
     }
 
     pub fn insert(&mut self, s: &'alloc str) -> SourceAtomSetIndex {
-        match self.atom_indices.get(s) {
-            Some(index) => return *index,
-            _ => {}
-        }
-
-        let index = self.atoms.len();
-        self.atoms.push(s);
-        let result = SourceAtomSetIndex::new(index);
-        self.atom_indices.insert(s, result);
-        result
+        let (index, _) = self.atoms.insert_full(s);
+        SourceAtomSetIndex::new(index)
     }
 
     pub fn get(&self, index: SourceAtomSetIndex) -> &'alloc str {
-        self.atoms[index.into_raw()].clone()
+        self.atoms.get_index(index.into_raw()).unwrap()
     }
 
-    pub fn into_vec(self) -> Vec<&'alloc str> {
-        self.atoms
+    pub fn into_vec(mut self) -> Vec<&'alloc str> {
+        self.atoms.drain(..).collect()
```

Comment:
nit: Maybe use `into_iter` instead of `drain`, as `into_iter` consume the content.

Why not using the `From` trait here?